### PR TITLE
Add ack_bundle_installation_warning step input

### PR DESCRIPTION
**Pull request by Bence1001**
### Checklist
- [x] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires _MINOR_ [version update](https://semver.org/)

### Context
* https://github.com/bitrise-steplib/steps-google-play-deploy/issues/121
* The installation of an App Bundle may be too large and trigger user warning on some Android devices.
* When an App Bundle like this is uploaded to Google Play this triggers a warning which needs to be explicitly acknowledged via passing an extra argument to the API/SDK.

### Changes
**Added new step input: Acknowledge Bundle Installation Warning (`ack_bundle_installation_warning`).**
* Default value: `false`.
* Passes `ack_bundle_installation_warning` argument to the Google Play SDK when uploading an App Bundle.
* Added extra error message about this input when this warning is found in the error returned by Google Play SDK.

### Investigation details
* We were unable to reproduce this warning with our own sample apps. The rules of this warning are not clearly documented, and it seems like it doesn't even trigger in all cases.

### Decisions
* We allow acknowledging the Bundle installation warning via a dedicated step input rather than hardcoding its value to `true`.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -13,18 +13,19 @@ import (
...
+	ExpansionfilePath            string          `env:"expansionfile_path"`
+	Track                        string          `env:"track,required"`
+	UserFraction                 float64         `env:"user_fraction,range]0.0..1.0["`
+	UpdatePriority               int             `env:"update_priority,range[0..5]"`
+	WhatsnewsDir                 string          `env:"whatsnews_dir"`
+	MappingFile                  string          `env:"mapping_file"`
+	ReleaseName                  string          `env:"release_name"`
+	Status                       string          `env:"status"`
+	RetryWithoutSendingToReview  bool            `env:"retry_without_sending_to_review,opt[true,false]"`
+	AckBundleInstallationWarning bool            `env:"ack_bundle_installation_warning,opt[true,false]"`
```
Comment (Bence1001) [inline config.go:28]: This is the new input, the rest is reformatting.

File: publish.go [hunk 2]
```diff
@@ -92,16 +95,21 @@ func uploadMappingFile(service *androidpublisher.Service, configs Configs, appEd
...
+	editsBundlesUploadCall.AckBundleInstallationWarning(ackBundleInstallationWarning)
 
 	bundle, err := editsBundlesUploadCall.Do()
 	if err != nil {
-		return &androidpublisher.Bundle{}, fmt.Errorf("failed to upload app bundle, error: %s", err)
+		msg := fmt.Sprintf("failed to upload app bundle, error: %s", err)
+		if strings.Contains(err.Error(), bundleInstallationWarning) {
+			msg = fmt.Sprintf("%s\nTo acknowledge this warning, set the Acknowledge Bundle Installation Warning (ack_bundle_installation_warning) input to true.", msg)
+		}
+		return &androidpublisher.Bundle{}, fmt.Errorf(msg)
```
Comment (godrei) [inline publish.go:112]: why don't we return `nil` instead of `&androidpublisher.Bundle{}` ?

File: main.go [hunk 3]
```diff
@@ -196,7 +196,7 @@ func executeEdit(service *androidpublisher.Service, configs Configs, changesNotS
 	log.Infof("Upload apks or app bundles")
 	versionCodes, err := uploadApplications(configs, service, appEdit)
 	if err != nil {
-		return fmt.Sprintf("Failed to upload APKs: %v", err)
+		return fmt.Sprintf("Failed to upload application(s): %v", err)
```
Comment (Bence1001) [inline main.go:199]: This error message was incorrect, because we might have been uploading an AAB.

**Review by godrei — APPROVED** [review]

**Comment by Bence1001** [comment]
Waiting for https://github.com/bitrise-steplib/steps-google-play-deploy/pull/122 to be merged before we can merge this.

**Review by ofalvai — APPROVED** [review]

**Review by zoltan-baba — APPROVED** [review]

**Pull request merged**